releases/v23_0_0.rst

```diff
@@ -1,7 +1,7 @@
 .. _release-v23-0-0:
 
 ###########################
-Release 23.0.0 (2021-12-21)
+Release 23.0.4 (2023-03-12)
```

Comment:
I just noticed this. Does this mean we're suppressing the original release date every time we do a new bug fix release? Normally the release log keeps a full history and I get that the Pipeline release documentation is a bit non standard for lots of reasons. I guess my question is: should we fix this issue?

Reply:
I will revert this then. Not sure this was intended or not updated in the past.

Reply:
Reverted